### ISSUE TITLE
[Error] Overfitting 발생 – 시퀀스 기반 학습 구성 이슈

### ISSUE THREAD

**Issue by SeBin7** [SeBin7/Road_Vision#18]
### **🔍 문제 요약**

- 학습 중 추론 결과가 고정되거나 특정 클래스(예: wet_road)로 편중되는 현상 발생.
- 최초에는 모델 출력 고정으로 오인되었으나, 훈련 단계에서 과적합이 빠르게 발생한 것이 원인으로 확인됨.

**Comment by SeBin7** [comment]
<img width="834" height="325" alt="Image" src="https://github.com/user-attachments/assets/01d714b6-8b64-4b56-9b8e-e08f8163d7b0" />

🔍 왜 이런 현상이 발생했는가?
1. 초기 과적합은 모델의 "가장 쉬운 클래스"에 집중
초기에 가장 눈에 띄는 클래스(예: 노면이 밝거나 물기가 뚜렷한 wet_road)만 잘 분류하게 됨

CrossEntropy는 맞춘 클래스에 대해서는 gradient를 거의 주지 않음 → 그 쪽으로 lock-in됨

2. 다른 클래스의 특징은 상대적으로 학습이 느림
normal_road, snow_road 등은 외형상 유사하고 경계도 모호함

→ 반복 학습하면서 점차 이 클래스들의 특징도 학습됨

3. Epoch이 진행되며 모델이 다양한 특징에 반응
MLP 레이어가 더 깊은 정보(분기, 차선 등)를 받아들이기 시작

→ softmax 분포가 한쪽으로 쏠리지 않게 됨
